documentation/src/main/asciidoc/topics/marshalling.adoc

```diff
@@ -141,11 +178,69 @@ include::code_examples/CustomMarshaller.java[]
 include::config_examples/custom_marshaller.xml[]
 ----
 
-
 TIP: It's possible custom marshaller implementations to access a configured white list via the
 link:{javadocroot}/org/infinispan/commons/marshall/Marshaller.html#initialize(org.infinispan.commons.configuration.ClassWhiteList)
 method, which is called during startup.
 
+== Adding Java Classes to Deserialization White Lists
+{brandname} does not allow deserialization of custom payloads into Java classes
+for security reasons, which applies to JSON, XML, and marshalled `byte[]`
+content.
+
+You must add Java classes to a deserialization white list, either using system
+properties or specifying them in the {brandname} configuration.
+
+.System properties
+
+[source,options="nowrap",subs=attributes+]
+----
+// Specify a comma-separated list of fully qualified class names
+-Dinfinispan.deserialization.whitelist.classes=java.time.Instant,com.myclass.Entity
+
+// Specify a regular expression to match classes
+-Dinfinispan.deserialization.whitelist.regexps=.*
+----
+
+.Declarative
+
+[source,xml,options="nowrap",subs=attributes+]
+----
+include::config_examples/deserialization_whitelist.xml[]
+----
+
+== Storing Deserialized Objects
```

Comment:
@ryanemerson I'm not sure about this section. lots of questions.

should we still refer to the Kryo and Protostuff marshallers as bridge implementations? 

what about the "bundle" jars? are these community only? I can't find them in the product downloads.

Reply:
> should we still refer to the Kryo and Protostuff marshallers as bridge implementations?

Nope



> what about the "bundle" jars? are these community only? I can't find them in the product downloads.

I don't think these marshaller implementations were ever productised. The "bundle" jars are still useful, as it provides the user with a single jar that they can just put in the server, opposed to having to add all the transitive dependencies of the Kryo or Protostuff marshaller.